File: ios/bugtags/BugtagsRN.m
//
//  BugtagsRN.h
//

#import "BugtagsRN.h"
#import <Bugtags/Bugtags.h>

@implementation BugtagsRN

RCT_EXPORT_MODULE()

RCT_EXPORT_METHOD(startWithAppKey:(NSString *)appKey invocationEvent:(NSInteger)invocationEvent)
{
  [Bugtags startWithAppKey:appKey invocationEvent:(BTGInvocationEvent)invocationEvent];
}

RCT_EXPORT_METHOD(setTrackingCrashes:(BOOL)trackingCrashes)
{
  [Bugtags setTrackingCrashes:trackingCrashes];
}

RCT_EXPORT_METHOD(setUserData:(NSString *)data forKey:(NSString *)key)
{
  [Bugtags setUserData:data forKey:key];
}

RCT_EXPORT_METHOD(removeUserDataForKey:(NSString *)key)
{
  [Bugtags removeUserDataForKey:key];
}

RCT_EXPORT_METHOD(removeAllUserData)
{
  [Bugtags removeAllUserData];
}

RCT_EXPORT_METHOD(addUserStep:(NSString *)content)
{
  [Bugtags addUserStep:content];
}

RCT_EXPORT_METHOD(log:(NSString *)content)
{
  [Bugtags log:content];
}

RCT_EXPORT_METHOD(sendFeedback:(NSString *)content)
{
  [Bugtags sendFeedback:content];
}

- (dispatch_queue_t)methodQueue
{
  return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: android/app/src/main/java/com/bugtagsrn/sdk/BugtagsRN.java
package com.bugtagsrn.sdk;

import android.app.Application;

import com.bugtags.library.Bugtags;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

/**
 * Created by kvh on 2018/5/14.
 */

public class BugtagsRN extends ReactContextBaseJavaModule {

    private ReactApplicationContext reactContext;

    public BugtagsRN(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "BugtagsRN";
    }

    @ReactMethod
    public void startWithAppKey(String appKey, int invocationEvent) {
        Bugtags.start(appKey, (Application) reactContext.getApplicationContext(), invocationEvent);
    }

    @ReactMethod
    public void setTrackingCrashes(boolean trackingCrashes) {
        Bugtags.setTrackingCrashes(trackingCrashes);
    }

    @ReactMethod
    public void setUserData(String data, String key) {
        Bugtags.setUserData(key, data);
    }

    @ReactMethod
    public void removeUserDataForKey(String key) {
        Bugtags.removeUserData(key);
    }

    @ReactMethod
    public void removeAllUserData() {
        Bugtags.removeAllUserData();
    }

    @ReactMethod
    public void addUserStep(String content) {
        Bugtags.addUserStep(content);
    }

    @ReactMethod
    public void log(String content) {
        Bugtags.log(content);
    }

    @ReactMethod
    public void sendFeedback(String content) {
        Bugtags.sendFeedback(content);
    }

}
